DS implementations/QUEUES/Queue_Implementation.cpp

```diff
@@ -0,0 +1,108 @@
+// C++ program for array implementation of queue
+
+#include <bits/stdc++.h>
+using namespace std;
+
+// A class to represent a queue
+class Queue
+{
+public:
+    int front, rear, size;
+    unsigned capacity;
+    int *array;
+};
+
+// function to create a queue of given capacity, it initializes size of queue as 0
+Queue *createQueue(unsigned capacity)
+{
+    Queue *queue = new Queue();
+    queue->capacity = capacity;
+    queue->front = queue->size = 0;
+
+    // assigning last index of the queue to the rear.
+    queue->rear = capacity - 1;
+    // dynamically allocating int array of capacity.
+    queue->array = new int[(queue->capacity * sizeof(int))]; 
+    return queue;
+}
+
+// Queue is full when size becomes equal to the capacity
+int isFull(Queue *queue)
+{
+    return (queue->size == queue->capacity);
+}
+
+// Queue is empty when size is 0
+int isEmpty(Queue *queue)
+{
+    return (queue->size == 0);
+}
+
+// Function to add an item to the queue. It changes rear and size
+void enqueue(Queue *queue, int item)
+{
+    // if queue is full then we can't push more.
+    if (isFull(queue))
+        return;
+    // updating rear to next index.
+    queue->rear = (queue->rear + 1) % queue->capacity;
+    // storing the item at the rear index of the queue.
+    queue->array[queue->rear] = item;
+    // as we added one more element so increment the size of the queue.
+    queue->size = queue->size + 1;
+    cout << item << " enqueued to queue\n";
+}
+
+// Function to remove an item from queue. It changes front and size
+int dequeue(Queue *queue)
```

Comment:
the variable name is not suitable. A simple `q`, `que` will be better. Even if code is compiling and running due to the scope of the variable. This will create a lot of confusion.